pkg/datamover/utils.go

```diff
@@ -0,0 +1,186 @@
+	"github.com/kanisterio/kanister/pkg/output"
+	"github.com/kanisterio/kanister/pkg/param"
+)
+
+const (
+	usePipeParam = `-`
+)
+
+func targetWriter(target string) (io.Writer, error) {
+	if target != usePipeParam {
+		return os.OpenFile(target, os.O_RDWR|os.O_CREATE, 0755)
+	}
+	return os.Stdout, nil
+}
+
+func locationPull(ctx context.Context, p *param.Profile, path string, target io.Writer) error {
+	return location.Read(ctx, target, *p, path)
+}
+
+// kopiaLocationPull pulls the data from a kopia snapshot into the given target
+func kopiaLocationPull(ctx context.Context, backupID, path, targetPath, password string) error {
+	switch targetPath {
+	case usePipeParam:
+		return snapshot.Read(ctx, os.Stdout, backupID, path, password)
+	default:
+		return snapshot.ReadFile(ctx, backupID, targetPath, password)
+	}
+}
+
+// connectToKopiaServer connects to the kopia server with given creds
+func connectToKopiaServer(ctx context.Context, kp *param.Profile) error {
+	contentCacheSize := kopia.GetDataStoreGeneralContentCacheSize(kp.Credential.KopiaServerSecret.ConnectOptions)
+	metadataCacheSize := kopia.GetDataStoreGeneralMetadataCacheSize(kp.Credential.KopiaServerSecret.ConnectOptions)
+	return repository.ConnectToAPIServer(
+		ctx,
+		kp.Credential.KopiaServerSecret.Cert,
+		kp.Credential.KopiaServerSecret.Password,
+		kp.Credential.KopiaServerSecret.Hostname,
+		kp.Location.Endpoint,
+		kp.Credential.KopiaServerSecret.Username,
+		contentCacheSize,
+		metadataCacheSize,
+	)
+}
+
+// connectToKopiaRepositoryServer connects to the kopia server with given repository server CR
+func connectToKopiaRepositoryServer(ctx context.Context, rs *param.RepositoryServer) (string, error) {
+	contentCacheMB, metadataCacheMB := kopiacmd.GetCacheSizeSettingsForSnapshot()
+	hostname, userPassphrase, certData, err := secretsFromRepositoryServerCR(rs)
+	if err != nil {
+		return "", errors.Wrap(err, "Error Retrieving Connection Data from Repository Server")
+	}
+	return userPassphrase, repository.ConnectToAPIServer(
+		ctx,
+		certData,
+		userPassphrase,
+		hostname,
+		rs.Address,
+		rs.Username,
+		contentCacheMB,
+		metadataCacheMB,
+	)
+}
+
+func secretsFromRepositoryServerCR(rs *param.RepositoryServer) (string, string, string, error) {
+	userCredJSON, err := json.Marshal(rs.Credentials.ServerUserAccess.Data)
+	if err != nil {
+		return "", "", "", errors.Wrap(err, "Error Unmarshalling User Credentials")
+	}
+	certJSON, err := json.Marshal(rs.Credentials.ServerTLS.Data)
+	if err != nil {
+		return "", "", "", errors.Wrap(err, "Error Unmarshalling Certificate")
+	}
+	hostname, userPassphrase, err := hostNameAndUserPassPhraseFromRepoServer(string(userCredJSON))
+	if err != nil {
+		return "", "", "", errors.Wrap(err, "Error Getting Hostname/User Passphrase from User credentials")
+	}
+	return hostname, userPassphrase, string(certJSON), err
+}
+
+func hostNameAndUserPassPhraseFromRepoServer(userCreds string) (string, string, error) {
```

Comment:
```suggestion
func hostnameAndUserPassphraseFromRepoServer(userCreds string) (string, string, error) {
```
[nit] hostname and passphrase are single words

Reply:
Done !